src/UI/UIView.cpp

```diff
@@ -0,0 +1,520 @@
+          {
+            char buf[32];
+            sprintf(buf, "##On%d%d", col, row);
+            bool b = m_selectedTrFn[row][col - 1];
+            ImGui::Checkbox(buf, &b);
+            m_selectedTrFn[row][col - 1] = b;
+            if (b)
+            {
+              std::cout << "ddd" << std::endl;
+            }
+          }
+
+        }
+      }
+
+      ImGui::EndTable();
+
+      ImGui::Checkbox("use transferfunction", &m_use_transferfunction);
+      if (m_use_transferfunction) {
+        bool is_multi_channel = m_controller_app.data_is_multi_channel();
+        if (is_multi_channel)
+        {
+          for (int i = 0; i < 3; i++) {
+            if (m_animated)
+            {
+              /*	
+                frame by frame animation
+
+                unsigned int active_volume = floor(m_frame);
+                unsigned int active_volume2 = ceil(m_frame);
+                double alpha = m_frame - active_volume;
+                if (active_volume < m_volumes[m_selectedVolume].size() && active_volume2 < m_volumes[m_selectedVolume].size())
+                {
+                  tfn_widget_multi[m_selectedVolume].setBlendedHistogram(
+                    m_volumes[m_selectedVolume][active_volume]->getTransferfunction(i),
+                    m_volumes[m_selectedVolume][active_volume2]->getTransferfunction(i), alpha, i);
+                }*/
+
+            }
+            else {
+              /*		tfn_widget_multi[m_selectedVolume].setHistogram(m_volumes[m_selectedVolume][0]->getTransferfunction(i), i);*/
+            }
+          }
+          m_controller_app.set_multi_transfer( true);
+          tfn_widget_multi[m_selectedTrnFnc].draw_ui();
+        }
+        else
+        {
+          if (m_animated)
+          {
+            /*	
+             frame by frame animation
+
+            unsigned int active_volume = floor(m_frame);
+              unsigned int active_volume2 = ceil(m_frame);
+              double alpha = m_frame - active_volume;
+              tfn_widget[m_selectedVolume].setMinMax(m_volumes[m_selectedVolume][active_volume]->getMin() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMin() * (1.0 - alpha),
+                m_volumes[m_selectedVolume][active_volume]->getMax() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMax() * (1.0 - alpha));
+              if (active_volume < m_volumes[m_selectedVolume].size() && active_volume2 < m_volumes[m_selectedVolume].size())
+                tfn_widget[m_selectedVolume].setBlendedHistogram(m_volumes[m_selectedVolume][active_volume]->getTransferfunction(0), m_volumes[m_selectedVolume][active_volume2]->getTransferfunction(0), alpha);*/
+
+          }
+          else if (numVolumes > 0) {
+            /*tfn_widget[m_selectedVolume].setHistogram(m_volumes[m_selectedVolume][0]->getTransferfunction(0));
+            tfn_widget[m_selectedVolume].setMinMax(m_volumes[m_selectedVolume][0]->getMin(), m_volumes[m_selectedVolume][0]->getMax());*/
+          }
```

Comment:
Likewise here, do you need this?

Reply:
Let's keep it for now. I'll remove it in the next PR. There are things I want to test without the need for rolling back specific code.